api/layer/system_object.go

```diff
@@ -68,53 +66,41 @@ func (n *layer) putSystemObjectIntoNeoFS(ctx context.Context, p *PutSystemObject
 	if err != nil && !errors.IsS3Error(err, errors.ErrNoSuchKey) {
 		return nil, err
 	}
-	idsToDeleteArr := updateCRDT2PSetHeaders(p.Metadata, versions, false) // false means "last write wins"
-
-	attributes := make([]*object.Attribute, 0, 3)
-
-	filename := object.NewAttribute()
-	filename.SetKey(objectSystemAttributeName)
-	filename.SetValue(p.ObjName)
 
-	createdAt := object.NewAttribute()
-	createdAt.SetKey(object.AttributeTimestamp)
-	createdAt.SetValue(strconv.FormatInt(time.Now().UTC().Unix(), 10))
-
-	versioningIgnore := object.NewAttribute()
-	versioningIgnore.SetKey(attrVersionsIgnore)
-	versioningIgnore.SetValue(strconv.FormatBool(true))
+	prm := PrmObjectCreate{
+		Container:  *p.BktInfo.CID,
+		Creator:    *p.BktInfo.Owner,
+		Attributes: make([][2]string, 2, 2+len(p.Metadata)),
+		Payload:    p.Reader,
+	}
 
-	attributes = append(attributes, filename, createdAt, versioningIgnore)
+	prm.Attributes[0][0], prm.Attributes[0][1] = objectSystemAttributeName, p.ObjName
+	prm.Attributes[1][0], prm.Attributes[1][1] = attrVersionsIgnore, "true"
 
 	for k, v := range p.Metadata {
-		attr := object.NewAttribute()
 		if !IsSystemHeader(k) {
 			k = p.Prefix + k
 		}
-		attr.SetKey(k)
-		if p.Prefix == tagPrefix && v == "" {
+
+		if v == "" && p.Prefix == tagPrefix {
 			v = tagEmptyMark
 		}
-		attr.SetValue(v)
-		attributes = append(attributes, attr)
-	}
 
-	raw := object.NewRaw()
-	raw.SetOwnerID(p.BktInfo.Owner)
-	raw.SetContainerID(p.BktInfo.CID)
-	raw.SetAttributes(attributes...)
+		prm.Attributes = append(prm.Attributes, [2]string{k, v})
+	}
 
-	ops := new(client.PutObjectParams).WithObject(raw.Object()).WithPayloadReader(p.Reader)
-	oid, err := n.pool.PutObject(ctx, ops, n.CallOptions(ctx)...)
+	id, err := n.neoFS.CreateObject(ctx, prm)
 	if err != nil {
 		return nil, n.transformNeofsError(ctx, err)
 	}
 
-	meta, err := n.objectHead(ctx, p.BktInfo.CID, oid)
+	meta, err := n.objectHead(ctx, p.BktInfo.CID, id)
 	if err != nil {
 		return nil, err
 	}
 
+	idsToDeleteArr := updateCRDT2PSetHeaders(p.Metadata, versions, false) // false means "last write wins"
```

Comment:
This line must be before forming attributes because `updateCRDT2PSetHeaders` method update `p.Metadata`